tests/integration/test_charm.py

```diff
@@ -78,6 +78,88 @@ async def test_setup_discourse(
     assert response.status_code == 200
 
 
+@pytest.mark.asyncio
+@pytest.mark.abort_on_fail
+async def test_db_migration(model: Model, ops_test: OpsTest, pytestconfig: Config, run_action):
+    """
+    arrange: preload postgres with a mock db
+    act: deploy and integrate with discourse
+    assert: discourse must be active idle
+    """
+    postgres_app = await model.deploy(
+        "postgresql-k8s",
+        channel="14/stable",
+        series="jammy",
+        revision=300,
+        trust=True,
+        config={"profile": "testing"},
+    )
+    async with ops_test.fast_forward():
+        await model.wait_for_idle(apps=[postgres_app.name], status="active")
+    # configure postgres
+    await postgres_app.set_config(
+        {
+            "plugin_hstore_enable": "true",
+            "plugin_pg_trgm_enable": "true",
+        }
+    )
+    await model.wait_for_idle(apps=[postgres_app.name], status="active")
+    db_pass = await run_action(postgres_app.name, "get-password", username="operator")
+    db_pass = db_pass["password"]
+    return_code, _, _ = await ops_test.juju(
+        "scp",
+        "--container",
+        # postgres_app.name,
+        "postgresql",
+        "./mock_db",
+        f"{postgres_app.units[0].name}:.",
+    )
+    assert return_code == 0
+
+    return_code, _, _ = await ops_test.juju(
+        "ssh",
+        "--container",
+        "postgresql",
+        postgres_app.units[0].name,
+        "createdb -h localhost -U operator --password discourse",
+        stdin=str.encode(f"{db_pass}\n"),
+    )
+    assert return_code == 0
+
+    return_code, _, _ = await ops_test.juju(
+        "ssh",
+        "--container",
+        "postgresql",
+        postgres_app.units[0].name,
+        "pg_restore -h localhost -U operator\
+              --password -d discourse\
+                  --no-owner --clean --if-exists ./mock_db",
+        stdin=str.encode(f"{db_pass}\n"),
+    )
```

Comment:
We're loading `./mock_db` which is a binary file, which makes it very hard to inspect/verify the contents. Would it either be possible to have that file be in a non-binary format, or can we do some verification in this test that the database is in the state we want it to be in (e.g. that it's from Discourse v3.2.0)?

Reply:
Thanks for pointing that out. It looks like I can do `pg_restore db.bin > db.sql` to make it human readable but I need to test if it is still importable in `.sql` format. For verification I can first install Discourse v3.2.0 to show that it works with the db then upgrade it to the current version. I did not do this to keep the test shorter and less resource hungry